test/test_linearizer.py

```diff
@@ -777,8 +777,9 @@ def get_recursive(uop): return set.union(set(uop.src), [uop], *[get_recursive(v)
     # check that the float4 cast collapses for all stores
     for store in local_stores+global_stores:
       assert store.src[2].dtype == dtypes.float.vec(2) and store.src[2].op is not UOps.CAST
-    # check the children's vins
-    assert barrier.src == tuple(local_stores)
```

Comment:
@Qazalin fyi, this was added 6 months ago. the comparison worked because we were not comparing ALU.